### ISSUE TITLE
[Registration] workflow improvement when registering using temporarily a second device: allow to use first verification code sent in first SMS

### ISSUE THREAD

**Issue by Wikinaut** [signalapp/Signal-Android#1399]
tl;dr
I propose to add a clean "registering a number by SMS to a different device" step.

User Story:

As a new user, when I want to register my telephone number B (SIM card for number B temporary inserted into my mobile phone B) on my tablet A (a device without phone or SMS capabilities) with the TextSecure PUSH server, I have to 
- enter the mobile number B into the TextSecure registration form on tablet A
- start the registration progress (my device B immediately receives (a first) 6-digit verification code SMS) [X]
- wait for the 2 minute time-out on tablet A (because A cannot receive the registration SMS)
- press the button on A for registration via an automated telephone voice call
- which now triggers the registration process by voice call to device B (user must be capable to understand a female voice, English language. This is a second and different verification code.

Problem:
- Not everyone is able to understand the quite quickly spoken verification code on the phone.
- Not everyone understands the English message
- Users cannot enter the _first_ verification code: because there is no possibility (no input form) in the current registration workflow
- Users have to know to wait for the timeout in order to trigger the "sent me code by voice call" process

Observation:
- I regard the **second SMS by voice as superfluous (in the standard case)**, because I already obtained a verification code on my device B at step [X], when I would only be able to enter this code on device A.

@moxie0 I think the verification workflow should be _deliberately modified_ which solves many issues: before starting the verification process at the device A, after having entered the telephone number, 

two new second buttons are proposed
- automatic registration (this is the present workflow with the silent SMS to the device where you press the button, but you have to enter the telephone (BTW: Can't this be fetched from the Android automatically?)
- **new button: "manual registration"** (a  verification code is sent via SMS to the device with the entered number)
- **new button: "registration by voice call"** (you receive an automated telephone call on the device with the entered number, and a six-digit verification code in English will be presented)

Advantages of my proposal:
- lower workload on TS registration server (the second SMS is not necessary, this step after the 2-minute-timeout is obsolete in the case that the user can receive the verification code on their second device)
- time saving: the user does not have wait for the 2-minute timeout
- easier: no need to understand the English voice message
- verification code text message (SMS) could be i10n/i18n
